### PR TITLE
BUGFIX: SD-425 Different reports in case of loading NULL fact

### PR DESCRIPTION
**Pull request by nadtan**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2209
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/1926

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3401/ : FAILURE in 2m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4215/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3402/ : SUCCESS in 7m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4216/ : SUCCESS in 17s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1811/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1161/ : SUCCESS in 20m 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4217/ : SUCCESS in 13s

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-939.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by naibinh** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1162/ : SUCCESS in 37m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4219/ : SUCCESS in 9s

File: src/components/visualizations/chart/chartOptions/extendedStackingChartOptions.ts [hunk 1]
```diff
@@ -36,21 +37,33 @@ export function getCategoriesForTwoAttributes(
...
             };
         },
         {},
     );
 
-    return Object.keys(combinedResult).map((name: string) => ({
-        name,
-        categories: combinedResult[name],
-    }));
+    return keys.map((key: string) => {
```
Comment (pbenes) [inline src/components/visualizations/chart/chartOptions/extendedStackingChartOptions.ts]: I'm just wondering if we need this `keys` array. I don't have a strong argument, but I feel that having .reduce() with side effects (pushing to keys array) is something we should not do.

What about just use `values` from lodash (https://lodash.com/docs/#values):
```
return values(combinedResult);
```
Then, the reduce would be without side effects and the keys array would not be needed.

Comment (nadtan) [reply src/components/visualizations/chart/chartOptions/extendedStackingChartOptions.ts]: The root cause of this bug comes from `Object.keys` issue, it returns an unexpected order. Assume we have an object like following:
```
const combinedResult = {
   "3": {},
   "2": {},
  "1": {}
};

Object.keys(combinedResult); // it returns ["1", "2", "3"] but the expected order is ["3", "2", "1"]
```
And unfortunately, `Object.values` or `values` from lodash that will return wrong order too.
That why I need `keys` array

Comment (pbenes) [reply src/components/visualizations/chart/chartOptions/extendedStackingChartOptions.ts]: Thanks for explanation

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1834/ : SUCCESS in 43m 33s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3450/ : SUCCESS in 8m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/937/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1003/ : SUCCESS in 43m 55s

**Pull request merged**